### PR TITLE
Use WFS API for road geometry

### PR DESCRIPTION
**Pull request by tatut**
The goal of this PR is to remove all direct road registry db dump uses from the pilot and replace them with WFS API use.

- New stored procedures that take a WKT geometry instead of road address
- New backend queries for doing road geometry extraction from WFS
- Frontend uses backend queries and fetches related features by passing the geometry
- Refactored the project geometry layers

### CODE REVIEW

File: app/backend/src/clj/teet/road/road_query.clj [hunk 1]
```diff
@@ -152,48 +152,63 @@
...
-                      error))
-      (-> body xml/parse zip/xml-zip
-          read-feature-collection))))
-
-(defn fetch-road-parts [{wfs-url :wfs-url} road-nr carriageway-nr]
-  (wfs-request wfs-url {:FILTER (query-by-road-and-carriageway road-nr carriageway-nr)}))
-
-(defn fetch-road-parts-by-coordinate [{wfs-url :wfs-url} coordinate distance-m]
-  (wfs-request wfs-url {:FILTER (query-by-coordinate coordinate distance-m)}))
+(defn- cached [cache-atom cache-key fetch-fn]
```
Comment (jkorpelin) [inline app/backend/src/clj/teet/road/road_query.clj]: Is this for development only? Since the `cache-options` is a commented atom.

Comment (tatut) [reply app/backend/src/clj/teet/road/road_query.clj]: Yes the cache options is for train development, you can fetch the items while in good network and have it cached.

The cache-atom is also used in THK import when updating multiple projects' geometries.

File: app/frontend/src/cljs/teet/project/project_controller.cljs [hunk 2]
```diff
@@ -406,25 +418,40 @@
...
+          ;; Fetch new road geometry if it hasn't been fetched yet or has changed
+          fetch-geometry? (or (not (contains? (get-in app [:route :project]) :geometry))
+                              (not= old-meter-range new-meter-range))]
+      (t/fx
+       (if fetch-geometry?
+         ;; Remove previously fetched geometry and road info
+         (update-in app [:route :project] dissoc :geometry :road-info)
+         app)
+
+       (when fetch-geometry?
```
Comment (jkorpelin) [inline app/frontend/src/cljs/teet/project/project_controller.cljs]: Should the `fetch-geometry` be it's own function?

File: app/backend/src/clj/teet/road/road_query.clj (hunk 1 above)
Comment (outamaa) [reply app/backend/src/clj/teet/road/road_query.clj]: Should rename and document the dev-only parts so that it's clear they're for dev use.

Comment (tatut) [reply app/backend/src/clj/teet/road/road_query.clj]: Caching is not a dev-only part, but I will document the cache-options in the queries

File: app/frontend/src/cljs/teet/project/project_controller.cljs [hunk 3]
```diff
@@ -406,25 +418,40 @@
...
-                    :carriageway carriageway}
-             :result-path [:route :project :basic-information-form :road-info]})))
+    (log/info "UpdateBasicInformationForm" form-data)
+    (let [project-meter-range (fn [app]
+                                (let [{[start-km-string end-km-string] :thk.project/km-range}
+                                      (get-in app [:route :project :basic-information-form])]
+                                  [(or (some-> start-km-string road-model/parse-km road-model/km->m)
+                                       (get-in app [:route :project :thk.project/start-m]))
+                                   (or (some-> end-km-string road-model/parse-km road-model/km->m)
+                                       (get-in app [:route :project :thk.project/end-m]))]))
```
Comment (outamaa) [inline app/frontend/src/cljs/teet/project/project_controller.cljs]: Not related specifically to this PR, more of a general observation: Getting the meter range from the app state has become pretty hideous.

File: app/frontend/src/cljs/teet/project/project_view.cljs [hunk 4]
```diff
@@ -133,7 +133,10 @@
 
 (defn project-map [e! {:keys [map page] :as app} project]
   (r/with-let [overlays (r/atom [])
-               set-overlays! #(reset! overlays %)
+               set-overlays! (fn [new-overlays]
+                               (when (not= new-overlays @overlays)
+                                 ;; Only set if actually changed (to avoid rerender loop)
+                                 (reset! overlays new-overlays)))
```
Comment (outamaa) [inline app/frontend/src/cljs/teet/project/project_view.cljs]: Sweet.

**Pull request merged**